File: Assets/Scripts/gamelogic/LevelInfo.js
#pragma strict

/**
 * @author: cmdr2 <secondary.cmdr2@gmail.com>
 */

private static var levelInfo : Array = Array(
	Array(3, Array("24", "30"), 0),
	Array(5, Array("24", "30"), 20),
	Array(7, Array("24", "30"), 30),
	Array(7, Array("24", "34"), 80),
	Array(11, Array("24", "31"), 110),
	Array(13, Array("25", "31"), 205),
	Array(14, Array("25", "35"), 300),
	Array(9, Array("20", "26"), 0),
	Array(15, Array("20", "26"), 40),
	Array(18, Array("21", "28"), 60),
	Array(20, Array("19", "32"), 160),
	Array(22, Array("18", "38"), 220),
	Array(24, Array("17", "40"), 410),
	Array(28, Array("20", "46"), 600)
);


public static function GetSpawnCount(levelNumber : int) : int {
	var l : Array = levelInfo[levelNumber - 1];
	return l[0];
}


public static function GetSpawnRadius(levelNumber : int) : Array {
	var l : Array = levelInfo[levelNumber - 1];
	return l[1];
}


public static function GetMagicDifficultyBonus(levelNumber : int) : int {
	var l : Array = levelInfo[levelNumber - 1];
	return l[2];
}

File: Assets/Scripts/gamelogic/LevelInfo.cs
using UnityEngine;
using System.Collections;

[System.Serializable]
public partial class LevelInfo : MonoBehaviour
{
    /**
 * @author: cmdr2 <secondary.cmdr2@gmail.com>
 */
    private static object[] levelInfo;
    public static int GetSpawnCount(int levelNumber)
    {
        object[] l = (object[]) LevelInfo.levelInfo[levelNumber - 1];
        return (int)l[0];
    }

    public static float[] GetSpawnRadius(int levelNumber)
    {
        object[] l = (object[]) LevelInfo.levelInfo[levelNumber - 1];
        return (float[])l[1];
    }

    public static int GetMagicDifficultyBonus(int levelNumber)
    {
        object[] l = (object[]) LevelInfo.levelInfo[levelNumber - 1];
        return (int)l[2];
    }

    static LevelInfo()
    {
        LevelInfo.levelInfo = new object[] {
            new object[] { 3, new float[] { 24, 30 }, 0 },
            new object[] { 5, new float[] { 24, 30 }, 20 },
            new object[] { 7, new float[] { 24, 30 }, 30 },
            new object[] { 7, new float[] { 24, 34 }, 80 },
            new object[] { 11, new float[] { 24, 31 }, 110 },
            new object[] { 13, new float[] { 25, 31 }, 205 },
            new object[] { 14, new float[] { 25, 35 }, 300 },
            new object[] { 9, new float[] { 20, 26 }, 0 },
            new object[] { 15, new float[] { 20, 26 }, 40 },
            new object[] { 18, new float[] { 21, 28 }, 60 },
            new object[] { 20, new float[] { 19, 32 }, 160 },
            new object[] { 22, new float[] { 18, 38 }, 220 },
            new object[] { 24, new float[] { 17, 40 }, 410 },
            new object[] { 28, new float[] { 20, 46 }, 600 }
        };
    }

}